src/Emulator/Peripherals/Peripherals/UART/STM32_UART.cs

```diff
@@ -91,7 +105,10 @@ private void DefineRegisters()
                 .WithTaggedFlag("NF", 2)
                 .WithFlag(3, FieldMode.Read, valueProviderCallback: _ => false, name: "ORE") // we assume no receive overruns
                 .WithTaggedFlag("IDLE", 4)
-                .WithFlag(5, out readFifoNotEmpty, FieldMode.Read | FieldMode.WriteZeroToClear, name: "RXNE") // as these two flags are WZTC, we cannot just calculate their results
+                .WithFlag(5, FieldMode.Read | FieldMode.WriteZeroToClear, name: "RXNE", valueProviderCallback: _ =>
+                {
+                    return readFifoNotEmpty;
```

Comment:
FYI:

This workaround I don't understand my self but otherwise, it is not working.

if _readFifoNotEmpty_ is a IFlagRegisterField, it happens that the machine does not see the change and the application stucks.

If we put inside the callback function, this is not happening anymore.